include/alpaka/math/fmod/FmodUniformCudaHipBuiltIn.hpp

```diff
@@ -34,23 +35,18 @@ namespace alpaka
                 FmodUniformCudaHipBuiltIn,
                 Tx,
                 Ty,
-                std::enable_if_t<std::is_floating_point<Tx>::value && std::is_floating_point<Ty>::value>>
+                std::enable_if_t<std::is_floating_point_v<Tx> && std::is_floating_point_v<Ty>>>
             {
                 __device__ auto operator()(FmodUniformCudaHipBuiltIn const& fmod_ctx, Tx const& x, Ty const& y)
                 {
                     alpaka::ignore_unused(fmod_ctx);
-                    return ::fmod(x, y);
-                }
-            };
-            //! The CUDA fmod float specialization.
-            template<>
-            struct Fmod<FmodUniformCudaHipBuiltIn, float, float>
-            {
-                __device__ auto operator()(FmodUniformCudaHipBuiltIn const& fmod_ctx, float const& x, float const& y)
-                    -> float
-                {
-                    alpaka::ignore_unused(fmod_ctx);
-                    return ::fmodf(x, y);
+
+                    if constexpr(is_decayed_v<Tx, float> && is_decayed_v<Ty, float>)
+                        return ::fmodf(x, y);
+                    else if constexpr(is_decayed_v<Tx, double> || is_decayed_v<Ty, double>)
+                        return ::fmod(x, y);
+                    else
+                        static_assert(!sizeof(Tx), "Unsupported data type");
```

Comment:
I find this a bit illogical. If implementation only supports arguments of the same type, why have them as `Tx`, `Ty` and not make this partial template specialization only for matching types? We still have a general version for all rest. Same applies for other binary operation changes in this PR

Reply:
Oh, I now see it actually supports mixed precision